components/supervisor/pkg/supervisor/supervisor.go

```diff
@@ -1554,6 +1555,48 @@ func analyseConfigChanges(ctx context.Context, wscfg *Config, w analytics.Writer
 	}
 }
 
+func trackReadiness(ctx context.Context, gitpodService *gitpod.APIoverJSONRPC, cfg *Config, cstate *InMemoryContentState, ideReady *ideReadyState, desktopIdeReady *ideReadyState) {
+	type SupervisorReadiness struct {
+		Kind                string `json:"kind,omitempty"`
+		WorkspaceId         string `json:"workspaceId,omitempty"`
+		WorkspaceInstanceId string `json:"instanceId,omitempty"`
+		Timestamp           int64  `json:"timestamp,omitempty"`
+	}
+	trackFn := func(ctx context.Context, gitpodService *gitpod.APIoverJSONRPC, cfg *Config, kind string) {
+		gitpodService.TrackEvent(ctx, &gitpod.RemoteTrackMessage{
```

Comment:
it does not return err or something like that to log?

Reply:
I print this error in a new commit, but when this reproduce, no error is returned and there is no useful information in the log, see `mustardmh-test-aabar7hcoyk`